### ISSUE TITLE
Nested subcommands with flags. Intermediate flags disappear.

### ISSUE THREAD

**Issue by sgirones** [codegangsta/cli#142]
When having more than one level of subcommands, flags are not accumulated.
Simple example:
```
package main

import (
	"fmt"
	"os"

	"github.com/codegangsta/cli"
)

func main() {
	app := cli.NewApp()
	app.Name = "redbooth"
	app.Flags = []cli.Flag{
		cli.StringFlag{"global-flag", "global-flag", "", ""},
	}
	app.Commands = []cli.Command{
		command1,
	}
	app.Run(os.Args)
}

var command1 = cli.Command{
	Name: "command1",
	Flags: []cli.Flag{
		cli.StringFlag{"command1-flag", "command1-flag", "", ""},
	},
	Subcommands: []cli.Command{
		subcommand1,
	},
}

var subcommand1 = cli.Command{
	Name: "subcommand1",
	Flags: []cli.Flag{
		cli.StringFlag{"subcommand1-flag", "subcommand1-flag", "", ""},
	},
	Action: func(c *cli.Context) {
		fmt.Println(c.GlobalString("global-flag"))
		fmt.Println(c.String("global-flag"))
		fmt.Println(c.GlobalString("command1-flag"))
		fmt.Println(c.String("command1-flag"))
		fmt.Println(c.GlobalString("subcommand1-flag"))
		fmt.Println(c.String("subcommand1-flag"))
	},
}

---

$ go run example.go --global-flag "global" command1 --command1-flag "cmd1" subcommand1 --subcommand1-flag "subcmd1"
global




subcmd1
```

I would expect command1-flag to be part of the globalSet, but currently is not appended.

**Comment by jszwedko** [comment]
This is an interesting question -- I /think/ the behaviour you describe is expected, but I'll have to ponder this a bit more.

**Comment by thorduri** [comment]
Didn't expect this behaviour, but came looking for methods to properly pass command flags to subcommands globally if you will.

I'd enjoy this behaviour from the package.
+1

**Comment by jszwedko** [comment]
Addressed by #227, let me know what you think!

**Comment by sgirones** [comment]
Looks good. Thanks for the PR!
